### ISSUE TITLE
we are plagued by "Repository ​​consistency ​​check ​​failed" errors

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#797]
I keep getting this error on both on our Jenkins (and Linux) and locally on my MacOS laptops.

example: https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3552/consoleFull

```
[error] The data already present in the dbuild repository for this data (uuid = ee88b93f53373fa312f12a92a44a84d0941a6af7)
[error] does not match the current metadata. This shouldn't happen! Please report.
[error] Key: meta/build/ee88b93f53373fa312f12a92a44a84d0941a6af7
```

the problem goes away if I `rm -rf ~/.dbuild`, but that's pretty rough because I lose all the cached runs

in 3+ years of community build work, I don't recall ever seeing this error before, but starting about a week or so ago, it started happening frequently

initially I thought it was because I had added a JDK 11 community build run to our Jenkins, so I figured it was crosstalk between the JDK 8, 10, and 11 runs sharing the same `~/.dbuild`. but I carefully altered our Jenkins config so that each of the three 2.12.x builds on the three JDK versions runs on a different worker, and blew away `~/.dbuild` on all three workers before triggering fresh runs. and to my surprise, the problem hasn't gone away.

@cunei any insight, can you help with this...?

**Comment by SethTisue** [comment]
(downgrading dbuild to 0.9.13 (on the off-chance it might help, I have no specific reason to believe it would) isn't an option because it doesn't work with sbt 1.2.1+)

**Comment by SethTisue** [comment]
I'm starting to wonder if this only happens when a run is aborted at an inopportune time, leaving files in an inconsistent state.

**Comment by cunei** [comment]
The Java version is (should) be calculated as part of the build hash, so in theory there shouldn't be collisions due to the same build done with different JDK versions.
This looks like a serialization/deserialization bug in dbuild. The differences are of the form:
```
< ..."cloc-plugin" % "0"), libraryDependencies += "javax.xml.bind" % "jaxb-api" % "2.3.0"))),List(
---
> ..."cloc-plugin" % "0")), List(libraryDependencies += "javax.xml.bind" % "jaxb-api" % "2.3.0"))),List(
```
Not sure why that started appearing just now, however; maybe some odd interaction due to a newer library being pulled in? I need to investigate. I'm on the move for the next few days, so I will look into it sometime next week.

**Comment by SethTisue** [comment]
@cunei surely this started with c28d3d2177f2578174e0b72078da161ec440a747, where I do:

```diff
+    extra.settings: ${vars.base.extra.settings} [
+      // for JDK9+. one-deeper nesting level makes it apply to build definition
+      ["libraryDependencies += \"javax.xml.bind\" % \"jaxb-api\" % \"2.3.0\""]
+    ]
```

this dbuild feature, of using the extra nesting level to make changes to settings for the build definition itself, is a dbuild feature I've never used before

**Comment by cunei** [comment]
I isolated the bug, it is indeed in the serialization/deserialization; will work on a fix. /cc @eed3si9n

**Comment by cunei** [comment]
This issue is fixed in https://github.com/lightbend/dbuild/commit/8827a458e43577cdaed8cce5dad362841974458f, I just released dbuild 0.9.15 with the fix. :blush:
/cc @eed3si9n

**Comment by SethTisue** [comment]
@cunei I upgraded all the builds (except 2.11/JDK6) to 0.9.15 and everything seems to be working fine. thank you very much!
